client/blocks/nps-survey/index.jsx

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import RecommendationSelect from './recommendation-select';
+
+class NpsSurvey extends Component {
+	static propTypes = {
+		onDismissed: PropTypes.func,
+		name: PropTypes.string
+	}
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			recommendationValue: null,
+			isSubmitting: false,
+			isSubmitted: false
+		};
+		this.handleRecommendationSelectChange = this.handleRecommendationSelectChange.bind( this );
+		this.handleFinishClick = this.handleFinishClick.bind( this );
+		this.handleDismissClick = this.handleDismissClick.bind( this );
+	}
+
+	handleRecommendationSelectChange( newRecommendationValue ) {
+		this.setState( {
+			recommendationValue: newRecommendationValue
```

Comment:
I guess we have different definitions of low :)

I consider the bit that does the writing of the state data to memory (what will get pulled out into the reducer) to be the lowest level code & would apply type guarantees there.

I don't feel strongly enough about it to block ya on it though.